code/modules/nanogate/nanogate.dm

```diff
@@ -59,9 +60,15 @@
 
 /obj/item/organ/internal/nanogate/removed_mob()
 	//Remove purchased powers
-	for (var/perk in perk_list)
-		if(ispath(perk, /datum/perk))
-			if(owner)
-				owner.stats.removePerk(perk)
-
+	SEND_SIGNAL(src, COMSIG_NANOGATE_REMOVED)
 	..()
+
+/obj/item/organ/internal/nanogate/proc/give_internal_computer()
+	var/obj/item/modular_computer/tablet/nanogate/I = new/obj/item/modular_computer/tablet/nanogate
```

Comment:
if emped this will have dire consequences

Reply:
I made that specific type of tablet uneffected by EMPs, though I think the code that makes it not turn on if the nanogate is broken might have slipped out while I was working on it.

Reply:
Okay, I've added a check to ensure that the nanogate isn't broken when you try to access the internal computer.